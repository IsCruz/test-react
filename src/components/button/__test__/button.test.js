// file for test case scenarios
import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Button from '../button';


it('render sin crashear el componente', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
});

it('el render debe de recibir el label como string', () => {
    // renders the button and expect to receive a testid value
    const { getByTestId } = render(<Button label='click' />);
    // call the id test and validate the props.label;
    expect(getByTestId('button')).toHaveTextContent('click');

});

it('El boton puede recibir numeros como label', () => {
    const { getByTestId } = render(<Button label={33} />);
    expect(getByTestId('button')).toBeInTheDocument(33);
});

// snapshot
it('El componente Button debe de hacer Match con el snapshot',() => {
    const template = renderer.create(<Button />).toJSON();
    expect(template).toMatchSnapshot();
});

it('El componente Button debe de hacer Match con el snapshot version 1',() => {
    const template = renderer.create(<Button label="version 1" />).toJSON();
    expect(template).toMatchSnapshot();
});
it('El componente Button debe de hacer Match con el snapshot version 2',() => {
    const template = renderer.create(<Button label="version 2" />).toJSON();
    expect(template).toMatchSnapshot();
});