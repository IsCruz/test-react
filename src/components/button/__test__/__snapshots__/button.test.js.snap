// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El componente Button debe de hacer Match con el snapshot 1`] = `
<div
  className="buttonGray"
  data-testid="button"
/>
`;

exports[`El componente Button debe de hacer Match con el snapshot version 1 1`] = `
<div
  className="buttonGray"
  data-testid="button"
>
  version 1
</div>
`;

exports[`El componente Button debe de hacer Match con el snapshot version 2 1`] = `
<div
  className="buttonGray"
  data-testid="button"
>
  version 2
</div>
`;
